### PR TITLE
feat: improve token handling and validation

### PR DESCRIPTION
**Pull request by JoaoPedroAS51**
Initialize Token and Refresh Token instances inside schemes. Both classes now use scheme options instead of `$auth`. Also move token and refresh token prefixes to scheme `DEFAULTS`.

Improve `check` method to also check tokens status, deprecating `_checkStatus`. Return object with token and refresh token expired status.
```js
{
  valid: boolean,
  tokenExpired: boolean,
  refreshTokenExpired: boolean,
  isRefreshable: boolean
}
```
**Note:** `valid` is the only required property.

Alias `HTTPRequest` to `AxiosRequestConfig` and `HTTPResponse` to `AxiosResponse`.

RequestHandler class now use scheme options and methods instead of `$auth`.

### CODE REVIEW

**Comment by DraftProducts** [comment]
Nice work

**Comment by JoaoPedroAS51** [comment]
Thanks 😄

**Comment by DraftProducts** [comment]
Have you defined the date of the V5 release ?

**Comment by JoaoPedroAS51** [comment]
Not yet. We want to make more improvements and refactors before release v5 :)

**Comment by DraftProducts** [comment]
You should maybe add a todo list to the README of the project, in this way interested developers could make their contribution 😝 

I would be interested myself 🙃

**Comment by JoaoPedroAS51** [comment]
Actually we have a board, but we need to add some more things to it.
https://github.com/nuxt-community/auth-module/projects/3

**Comment by JoaoPedroAS51** [comment]
[Improve token handling and validation](https://trello.com/c/67y03LkJ/1-improve-token-handling-and-validation)

File: src/core/auth.ts [hunk 1]
```diff
@@ -1,7 +1,6 @@
+import type { AxiosRequestConfig, AxiosResponse } from 'axios'
```
Comment (pi0) [inline src/core/auth.ts:1]: Even if current request adaptor is axios, isn't it better to keep abstraction from types? (alias HTTPRequest to AxiosRequestConfig, etc)

File: src/inc/request-handler.ts [hunk 2]
```diff
@@ -1,35 +1,45 @@
...
 
 export default class RequestHandler {
-  constructor (public $auth: Auth) {}
+  public scheme: Scheme
+  public $axios: NuxtAxiosInstance
+  public interceptor: any
+
+  constructor (scheme, axios) {
+    this.scheme = scheme
+    this.$axios = axios
```
Comment (pi0) [inline src/inc/request-handler.ts:13]: $scheme is now scheme :D Why not doing same for axios?

File: src/schemes/local.ts [hunk 3]
```diff
@@ -36,39 +40,61 @@ const DEFAULTS: SchemeOptions = {
...
+      return true
+    }
+
+    // Get status
+    const tokenStatus = this.token.status()
+
+    // Token has expired. Attempt `tokenCallback`
+    if (tokenStatus.expired()) {
+      if (typeof tokenCallback === 'function') {
+        return tokenCallback(false) || false
```
Comment (pi0) [inline src/schemes/local.ts:83]: We can instead, return either false or false or an object like `{ tokenExpired: true }` so check is still not-falsy but can provide better meta.

File: src/inc/refresh-token.ts [hunk 4]
```diff
@@ -1,37 +1,40 @@
...
 
 export default class RefreshToken {
-  public $auth: Auth
+  public scheme: Scheme
+  public $storage: Storage
 
-  constructor (auth: Auth) {
-    this.$auth = auth
+  constructor (scheme: Scheme, storage: Storage) {
+    this.scheme = scheme
```
Comment (pi0) [inline src/inc/refresh-token.ts:12]: However we call scheme instance strategy, i like your idea for simplification. Maybe can always call it scheme :+1:

File: src/schemes/local.ts [hunk 5]
```diff
@@ -36,39 +40,61 @@ const DEFAULTS: SchemeOptions = {
...
-      this.$auth.token.set(token)
+      this.token.set(token)
     }
   }
 
-  _checkStatus () {
-    if (this.options.token.required) {
-      // Sync token
-      this.$auth.token.sync()
+  check (checkStatus = false, tokenCallback?, _refreshTokenCallback?) {
```
Comment (pi0) [inline src/schemes/local.ts:63]: This callback-based approach is adding again a dependency complexity and making function, not pure check. What if we instead, always return an object like `{ valid: bool, tokenExpired: true }` this way consumers can take extra actions based on validation feedback.

Comment (JoaoPedroAS51) [reply src/schemes/local.ts]: Improved in commit 4d014e13

File: src/schemes/local.ts (hunk 3 above)
Comment (JoaoPedroAS51) [reply src/schemes/local.ts]: Improved in commit 4d014e13

File: src/inc/request-handler.ts (hunk 2 above)
Comment (JoaoPedroAS51) [reply src/inc/request-handler.ts]: Renamed in commit 91a2fd03

File: src/core/auth.ts (hunk 1 above)
Comment (JoaoPedroAS51) [reply src/core/auth.ts:1]: Yeah, I agree!

Comment (JoaoPedroAS51) [reply src/core/auth.ts]: Refactored in commit 1171a8ac

**Pull request merged**